morango/models/core.py

```diff
@@ -115,49 +113,55 @@ class InstanceIDModel(UUIDModelMixin):
     system_id = models.CharField(max_length=100, blank=True)
 
     @classmethod
+    @transaction.atomic
```

Comment:
The changes in the `get_or_create_current_instance` method, below, implement the behavior described in  https://github.com/learningequality/morango/issues/79 (leveraging the helper functions in `utils.py`)

First, it sees whether the current system parameters (calculated using the "old method") match the "current" instance. If they do, we stick with that to minimize unnecessary churn. If not, we use the "new method" to compute a more robust instance ID.